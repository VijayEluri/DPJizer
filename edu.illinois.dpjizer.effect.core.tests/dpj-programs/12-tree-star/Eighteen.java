/**
 * This file is licensed under the University of Illinois/NCSA Open Source License. See LICENSE.TXT for details.
 */

/**
 * 
 * @author Mohsen Vakilian
 * 
 */
class Node<region P> {
	region L, R;
	double mass in P;
	Node<L> left in L;
	Node<R> right in R;

	Node() {}
	
	void setMass(double mass) {
		this.mass = mass;
	}
	
	void setMassOfChildren(double mass) {
		cobegin {
			if (left != null) left.setMass(mass);
			if (right != null) right.setMass(mass);
		}
	}
}