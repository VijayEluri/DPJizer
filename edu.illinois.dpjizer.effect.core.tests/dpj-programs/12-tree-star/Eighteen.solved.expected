Node.Node() pure
Node.setMass(double) writes P
Node.setMassOfChildren(double) writes Root : Node.L, Root : Node.R
